target/linux/x86/config-5.15

```diff
@@ -221,7 +221,7 @@ CONFIG_MEMFD_CREATE=y
 CONFIG_MICROCODE=y
 CONFIG_MICROCODE_AMD=y
 CONFIG_MICROCODE_INTEL=y
-CONFIG_MICROCODE_OLD_INTERFACE=y
+# CONFIG_MICROCODE_LATE_LOADING is not set
```

Comment:
Are you sure we are not using this late interface?

I do not know how the microcode loading in OpenWrt works.

Reply:
I am not sure but the symbol has been removed upstream.  As you suggesting that we set `CONFIG_MICROCODE_LATE_LOADING=y`?  There is a prominent warning in the help against it.

Reply:
I do not know. Does someone have an x86 system, with OpenWrt and could check how we install the microcode updates please.

Reply:
@hauke 

I have, but I do early loading.
Here's microcode documentation though: https://www.kernel.org/doc/html/next/x86/microcode.html

and a summary..

Early loading:
microcodes must be on cpio namespace (initrd/initramfs) and located flowingly (mine though differed from this..)
 - on Intel: kernel/x86/microcode/GenuineIntel.bin
 - on AMD: kernel/x86/microcode/AuthenticAMD.bin

Late loading:
`echo 1 > /sys/devices/system/cpu/microcode/reload` triggers loading.
> The loading mechanism looks for microcode blobs in /lib/firmware/{intel-ucode,amd-ucode}

And here's some reading about the *change* you mentioned https://www.phoronix.com/news/Linux-5.19-Late-ucode-Loading

Reply:
OpenWrt is sort of special and does not use an initramfs, but it could be that the rootfs is already available early enough. I do not know how the boot process on x86 works in detail.

Reply:
We use the late loading, see https://github.com/openwrt/openwrt/blob/master/target/linux/x86/base-files/lib/preinit/02_load_x86_ucode
We should activate `CONFIG_MICROCODE_LATE_LOADING` for x86.

Reply:
@hauke

rootfs available early or not does not change that any other than cpio is unavailable option for early loading of ucode. I also tested, ucode on rootfs (ext4), does not get loaded at all on any point.

And sure openwrt uses initramfs; or at least supports it, its basic option in kernel and enabled as default, and it should. Openwrt though doesn't provide ucode **as** initramfs- but ucode is necessary **only** for x86_64, so there are variety of boot loaders available for it - and afaik **all** of them support initrd/initramfs.
And even if they wouldn't; just put initrd=intel-ucode.bin to your kernel cmd-line and you probably are all set with just that.

ucode though isn't provided as cpio archive on openwrt, if you meant that; I don't know what should be done for that. On mine, I mount bios boot to /boot and EFI Boot support to /boot/EFI - so ucode goes to /boot... But this isn't _set_ directory structure, so where to extract package containing ucode then? It takes about 6mb if I remember correctly so wouldn't be that interested of storing a backup of it in.. let's say /usr/share/ucode for example. Even though x86_64 systems usually have modular storage options that allow quite large "roms" (hard disks) where 6mb should not be an issue.

ucode initramfs does not need to have init or anything else besides firmware in preset path.

Reply:
Thanks for the discussion and for the digging you two.  I enabled it and added some verbiage to the commit message doing so.